app/src/test/java/com/antosik/benchproject/app/common/binding/recyclerView/RecyclerViewBindingAdaptersKtTest.kt

```diff
@@ -0,0 +1,31 @@
+package com.antosik.benchproject.app.common.binding.recyclerView
+
+import androidx.recyclerview.widget.RecyclerView
+import com.antosik.benchproject.app.common.recyclerView.RecyclerBaseAdapter
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import org.junit.jupiter.api.Test
+
+internal class RecyclerViewBindingAdaptersKtTest {
+    val recyclerBaseAdapter: RecyclerBaseAdapter<Any> = mockk(relaxed = true)
+    val recyclerView: RecyclerView = mockk {
+        every { adapter } returns recyclerBaseAdapter
+    }
+
+    @Test
+    fun `SHOULD invoke method uploadData WHEN input data is NOT null`() {
+        val data: List<Any> = mockk(relaxed = true)
+
+        recyclerView.updateAdapterData(data)
+
+        verify { recyclerBaseAdapter.uploadData(data) }
+    }
+
+    @Test
+    fun `SHOULD don't invoke method uploadData WHEN input data is null`() {
+        recyclerView.updateAdapterData(null)
+
+        verify(exactly = 0) { recyclerBaseAdapter.uploadData(any()) }
```

Comment:
I tried to make this tests dynamics but when I did this I had to use data or any() in verify (it depended if data will be null or not). Therefore in my opinion two tests will be ok instead of adjusting my tests only to use dynamics tests 😄